DataSetExplorer/DataSetBuilder/DataSetAnalysisService.cs

```diff
@@ -30,16 +40,39 @@ public Result<string> FindInstancesRequiringAdditionalAnnotation(string dataSetP
                 var exporter = new TextFileExporter(outputPath);
                 exporter.ExportInstancesWithAnnotatorId(dataset.GetInsufficientlyAnnotatedInstances());
                 return Result.Ok("Instances requiring additional annotation exported: " + outputPath);
-            } catch (IOException e)
+            }
+            catch (IOException e)
             {
                 return Result.Fail(e.ToString());
             }
         }
 
+        public Result<List<DataSetInstance>> FindInstancesWithAllDisagreeingAnnotations(int dataSetId)
+        {
+            var dataset = LoadDataSet(dataSetId);
+            if (dataset == default) return Result.Fail($"DataSet with id: {dataSetId} does not exist.");
+            var instances = dataset.GetInstancesWithAllDisagreeingAnnotations();
+            return Result.Ok(instances);
+        }
+
+        public Result<List<DataSetInstance>> FindInstancesRequiringAdditionalAnnotation(int dataSetId)
+        {
+            var dataset = LoadDataSet(dataSetId);
+            if (dataset == default) return Result.Fail($"DataSet with id: {dataSetId} does not exist.");
+            var instances = dataset.GetInsufficientlyAnnotatedInstances();
+            return Result.Ok(instances);
+        }
+
         private DataSet LoadDataSet(string folder)
         {
             var importer = new ExcelImporter(folder);
             return importer.Import("Clean CaDET");
         }
+
+        private DataSet LoadDataSet(int dataSetId)
```

Comment:
No need for this method, as you've added 5 lines of code to replace _repository.GetDataSet(dataSetId) with GetDataSet(dataSetId). Furthermore, it is not clear how LoadDataSet(string) is different from LoadDataSet(int) from the function headers.